src/app/templates/activity/Activity.tsx

```diff
@@ -1,188 +1,88 @@
-  address: string;
+interface ActivityCompProps {
   assetSlug?: string;
-  operations?: IOperation[];
-  hasMoreRef: React.MutableRefObject<boolean>;
 }
 
-const handleLoadMore = async ({
-  setLoadingMore,
-  setSyncError,
-  setRestOperations,
-  chainId,
-  address,
-  assetSlug,
-  operations,
-  hasMoreRef
-}: HandleLoadMore) => {
-  setLoadingMore(true);
-
-  try {
-    await syncOperations('old', chainId, address);
-  } catch (err: any) {
-    console.error(err);
-    setSyncError(err);
-  }
-
-  try {
-    const oldOperations = await fetchOperations({
-      chainId,
-      address,
-      assetIds: assetSlug ? [assetSlug] : undefined,
-      limit: ACTIVITY_PAGE_SIZE,
-      offset: operations?.length ?? 0
-    });
-    if (oldOperations.length === 0) {
-      hasMoreRef.current = false;
-    }
-
-    setRestOperations((ops: IOperation[]) => [...ops, ...oldOperations]);
-  } catch (err: any) {
-    console.error(err);
+export const ActivityComponent: React.FC<ActivityCompProps> = ({ assetSlug }) => {
+  const {
+    loading,
+    reachedTheEnd,
+    list: activities,
+    loadMore: loadMoreActivities
+  } = useActivities(INIT_OPERS_N, assetSlug);
+
+  const { publicKeyHash: currentAccountAddress } = useAccount();
+  const chainId = useChainId(true);
+  const syncSupported = useMemo(() => isKnownChainId(chainId), [chainId]);
+
+  const onLoadMoreBtnClick = () => {
+    loadMoreActivities(OPERS_LOAD_STEP);
+  };
+
+  const onRetryLoadBtnClick = () => {
+    loadMoreActivities(INIT_OPERS_N);
+  };
+
+  if (activities.length === 0) {
+    if (loading) return <ActivitySpinner height="2.5rem" />;
+    else if (reachedTheEnd === false)
+      return (
+        <div className="w-full flex justify-center mt-5 mb-3">
+          <FormSecondaryButton onClick={onRetryLoadBtnClick} small>
+            <T id="tryLoadAgain" />
+          </FormSecondaryButton>
+        </div>
+      );
+
+    return (
+      <div className={classNames('mt-4 mb-12', 'flex flex-col items-center justify-center', 'text-gray-500')}>
+        <LayersIcon className="w-16 h-auto mb-2 stroke-current" />
+
+        <h3 className="text-sm font-light text-center" style={{ maxWidth: '20rem' }}>
+          <T id="noOperationsFound" />
+        </h3>
+      </div>
+    );
   }
 
-  setLoadingMore(false);
+  return (
+    <>
+      <div className={classNames('w-full max-w-md mx-auto', 'flex flex-col')}>
+        {activities.map(activity => (
+          <ActivityItemComp
+            key={activity.hash}
+            address={currentAccountAddress}
+            activity={activity}
+            syncSupported={syncSupported}
+          />
+        ))}
+      </div>
+
+      {loading ? (
+        <ActivitySpinner height="2.5rem" />
+      ) : reachedTheEnd === false ? (
+        <div className="w-full flex justify-center mt-5 mb-3">
+          <FormSecondaryButton onClick={onLoadMoreBtnClick} small>
+            <T id="loadMore" />
+          </FormSecondaryButton>
+        </div>
```

Comment:
remove this button and add [infinite loading list](https://github.com/bvaughn/react-virtualized/blob/HEAD/docs/creatingAnInfiniteLoadingList.md)